### PR TITLE
Fixed#2642: Convert py module references to six module

### PR DESCRIPTION
**Pull request by srinivasreddy**

### CODE REVIEW

**Comment by The-Compiler** [comment]
This should definitely be done on the `features` branch and not `master`.

File: _pytest/nose.py [hunk 1]
```diff
@@ -66,7 +66,7 @@ def is_potential_nosetest(item):
 def call_optional(obj, name):
     method = getattr(obj, name, None)
     isfixture = hasattr(method, "_pytestfixturefunction")
-    if method is not None and not isfixture and py.builtin.callable(method):
+    if method is not None and not isfixture and six.callable(method):
```
Comment (RonnyPfannschmidt) [inline _pytest/nose.py]: i believe the callable builtin is back on all supported python versions so should be able drop the six reference here

File: setup.py [hunk 2]
```diff
@@ -43,7 +43,7 @@ def has_environment_marker_support():
 
 
 def main():
-    install_requires = ['py>=1.4.33', 'setuptools']  # pluggy is vendored in _pytest.vendored_packages
+    install_requires = ['py>=1.4.33', 'six','setuptools']  # pluggy is vendored in _pytest.vendored_packages
```
Comment (RonnyPfannschmidt) [inline setup.py]: i do wonder if we should have a minimal version there,
i suspect its necessary once we sort out `_pytest.compat`

File: _pytest/nose.py (hunk 1 above)
Comment (The-Compiler) [inline _pytest/nose.py]: Yep, from the six docs:

> Note callable has returned in Python 3.2, so using six’s version is only necessary when supporting Python 3.0 or 3.1.

File: _pytest/python.py [hunk 3]
```diff
@@ -613,7 +613,7 @@ def getcallargs(self, obj):
         if not isinstance(obj, (tuple, list)):
             obj = (obj,)
         # explicit naming
-        if isinstance(obj[0], py.builtin._basestring):
+        if isinstance(obj[0], six.string_types):
```
Comment (RonnyPfannschmidt) [inline _pytest/python.py]: i just noticed the missing import here, please add it

File: setup.py [hunk 4]
```diff
@@ -43,7 +43,7 @@ def has_environment_marker_support():
 
 
 def main():
-    install_requires = ['py>=1.4.33', 'setuptools']  # pluggy is vendored in _pytest.vendored_packages
+    install_requires = ['py>=1.4.33', 'six>=1.10.0','setuptools']  # pluggy is vendored in _pytest.vendored_packages
```
Comment (The-Compiler) [inline setup.py]: Why no older versions?

**Comment by RonnyPfannschmidt** [comment]
well done, after the last import linting issue is resolved, we can take a look at vendoring six in order to protect ourselves against certain scenarios

**Comment by nicoddemus** [comment]
Thanks @srinivasreddy! 

The only thing missing is a changelog entry, I suggest `changelog/2642.trivial`:

```
Refactored internal Python 2/3 compatibility code to use ``six``.
```

**Comment by nicoddemus** [comment]
Btw, we can merge this as is and decide later if we need to vendor `six` or not.

**Comment by srinivasreddy** [comment]
I think i am done with this PR. Whatever changes needed to remove py lib completely, can be done in another PR. :+1:

**Comment by nicoddemus** [comment]
> I think i am done with this PR. Whatever changes needed to remove py lib completely, can be done in another PR.

Definitely, we appreciate it! Just pushed the changelog myself, I think this can be merged.

Thanks again @srinivasreddy!

**Comment by RonnyPfannschmidt** [comment]
@srinivasreddy well done, good work and thank you for preserving 👍

**Pull request merged**